src/web-components/SelectWallet/index.ts

```diff
@@ -0,0 +1,117 @@
+import { LitElement, html, css } from 'lit';
+import { customElement, state, property } from 'lit/decorators.js';
+import { SelectWalletStyles } from './styles';
+
+/**
+ * `select-wallet` is a custom LitElement component for selecting a cryptocurrency wallet.
+ * 
+ * ### Slots:
+ * - **slottedButtons**: Slot for action buttons (e.g., confirm or cancel buttons).
+ * 
+ * ### State:
+ * - `selectedWallet` (string): Tracks the selected wallet key.(set to currently selected wallet for active state)
+ * 
+ * ### Properties:
+ * - `slottedButtonsStyling` (string): Styles the slotted buttons container.
+ * - `addFunction` (Function): Function that runs when the component is first updated (`firstUpdated()`).
+ * - `infiniteScroll` (Function): Function to handle infinite scrolling within the wallet list.
+ * - `selectWallet` (Function): Function to handle wallet selection. Receives the wallet object.
+ * - `wallets` (Array): An array of wallet objects, each with `key`, `name`, and `icon` properties.
+ * 
+ * ### Methods:
+ * - `renderSelectIcon(walletKey: string)`: Renders a selection icon for the wallet.
+ * 
+ * ### Example usage:
+ * 
+ * ```html
+ * <select-wallet
+ *   .wallets="${wallets}"
+ *   .selectWallet="${onSelectWallet}">
+ * </select-wallet>
+ * ```
+ * 
+ * ```javascript
+ * const wallets = [
+ *   { key: 'lace', name: 'Lace', icon: 'path/to/lace-icon.svg' },
+ *   { key: 'nami', name: 'Nami', icon: 'path/to/nami-icon.svg' },
+ * ];
+ * 
+ * function onSelectWallet(wallet) {
+ *   console.log('Selected wallet:', wallet);
+ * }
+ * ```
+ */
+
+@customElement('select-wallet')
+export class SelectWallet extends LitElement {
+    @state() selectedWallet: string = '';
+    @state() walletApi: any = null;
+
+    @property({ type: String }) route = '';
+    @property({ type: String }) help = 'help';
+    @property({ type: String }) slottedButtonsStyling = '';
+    @property({ type: Function }) addFunction = () => { };
+    @property({ type: Function }) infiniteScroll = () => { };
+    @property({ type: Function }) selectWallet = (wallet) => { };
+    @property({ type: Array }) wallets = [];
+
+    static styles = SelectWalletStyles;
+
+    firstUpdated() {
+        this.addFunction();
+    }
```

Comment:
what is this for?

Reply:
this lets a dev add a function to fire inside the component if they need to.

Reply:
like if they wanted to make sure that a properties was properly appearing inside of if they wanted to call/update the properties when the component renders, it isn't truly necessary i just didn't like cutting the dev off from the component completely and also wanted to provide a potential fix to unforeseen use cases